draft-ietf-quic-multipath.md

```diff
@@ -587,26 +579,25 @@ the ACK frame as well as packet protection as described in the following subsect
 The ACK_MP frame, as specified in {{ack-mp-frame}}, is used to
 acknowledge 1-RTT packets.
 Compared to the QUIC version 1 ACK frame, the ACK_MP frame additionally
-contains a Path ID.
-The Path ID is used to distinguish packet number spaces for different
-paths and is simply derived from the sequence number of
-Destination Connection ID.
-Therefore, the Path ID for 1-RTT packets can be identified
-based on the Destination Connection ID in each packet.
+contains the Receiver's Destination Connection ID Sequence Number field
+to distinguish the Connection ID-specific packet number space
+acknowledged.
 
 Acknowledgements of Initial and Handshake packets MUST be carried using
 ACK frames, as specified in {{QUIC-TRANSPORT}}. The ACK frames, as defined
-in {{QUIC-TRANSPORT}}, do not carry path identifiers. If for any reason
-ACK frames are received in 1-RTT packets while the state of multipath
-negotiation is ambiguous, they MUST be interpreted as acknowledging
-packets sent on path 0.
+in {{QUIC-TRANSPORT}}, do not carry the Receiver's Destination
+Connection ID Sequence Number field. If multipath has been successfully
+negotiated, ACK frames in 1-RTT packets acknowledge packets sent with
+the Connection ID having sequence number 0.
 
 As soon as the negotiation of multipath support is completed,
 endpoints SHOULD use ACK_MP frames instead of ACK frames to acknowledge application
-data packets, including 0-RTT packets, received on path ID 0 after the handshake concluded.
+data packets, including 0-RTT packets, using the Connection ID with
+sequence number 0 after the handshake concluded.
 
-ACK_MP frame (defined in {{ack-mp-frame}}) SHOULD be sent on the same path
-as identified by the Path Identifier. However, an ACK_MP frame can be returned via a
+ACK_MP frame (defined in {{ack-mp-frame}}) SHOULD be sent on the path
+it perceives the Connection ID of the packet number space it acknowledges.
```

Comment:
```suggestion
ACK_MP frame (defined in {{ack-mp-frame}}) SHOULD be sent on same the path
as the data was sent that it acknowledged.
```

Reply:
While I understand the suggestion, I'm not sure this makes the text clearer, though I don't have strong opposition here.

Reply:
I fully agree here. Sending the ACK_MP on the same path is a reasonable default behaviour, but in some network configurations, this is clearly suboptimal and there is no point forcing endpoints to do that. I think having a  "SHOULD" is fine, but not a "MUST".

Rethinking further, I'm wondering if this paragraph could even be removed based on discussions in #183. We already discuss the ACK_MP scheduling in implementation considerations, and that should be enough. Maybe add at some place that frames can effectively be sent and received on any valid/active path.

Reply:
```suggestion
ACK_MP frame (defined in {{ack-mp-frame}}) SHOULD be sent on the path
it received packet with the Connection ID of the packet number space it acknowledges.
```

Reply:
On the editorial part: I think "perceives" is just the wrong word. I was also trying to make it shorter but feel free to take any of the two suggestions.

On the text itself, if we use normative language it should stay here and not move to the implementation guidance section. let keep this for now and maybe open a separate issue in case we need more discussion or actually want to change it.

My view is: I think we need some normative guidance because of the RTT calculation issue. However, there might be room for refining it or provide further explanation in the implementation guidance scheduling section.

Reply:
"receive packet with" is a better text than "perceived".

And yes, let's merge this PR as is, and discuss path affinity of MP_ACK separately.

Reply:
I started a discussion of the "ACK on same path" in issue #190